File: core/src/main/java/com/fahamutech/kcore/admin/database/connector/TestimonyDataSource.java
package com.fahamutech.kcore.admin.database.connector;

import com.fahamutech.kcore.admin.database.DataBaseCallback;
import com.fahamutech.kcore.admin.model.ITestimony;

public interface TestimonyDataSource {

    void createTestimony(ITestimony testimony, DataBaseCallback... callbacks);

    void deleteTestimony(String docId, DataBaseCallback... callbacks);
}

File: core/src/main/java/com/fahamutech/kcore/admin/database/connector/TestimonyDataSource.kt
package com.fahamutech.kcore.admin.database.connector

import com.fahamutech.kcore.admin.database.DataBaseCallback
import com.fahamutech.kcore.admin.model.ITestimony

interface TestimonyDataSource {
    fun createTestimony(testimony: ITestimony?, vararg callbacks: DataBaseCallback?)
    fun deleteTestimony(docId: String?, vararg callbacks: DataBaseCallback?)
}